### ISSUE TITLE
About Autograd Support in Ftorch

### ISSUE THREAD

**Issue by nckxneoro52449** [Cambridge-ICCS/FTorch#131]
I have a question regarding the functionality of Ftorch. Does Ftorch currently support PyTorch's autograd feature for calculating first-order derivatives (Jacobians)? If not, are there any plans to add this feature in the future?

**Comment by jatkinson1000** [comment]
At present we do not include this functionality, but there are active plans to do so in future: See e.g. https://github.com/Cambridge-ICCS/FTorch/discussions/22

It is something we hope to look at soon after we finalise GPU functionalities make a first release, but community contributions are also welcome.
Even if it is just a list of the functionalities you would like to see made available through FTorch.

**Comment by jatkinson1000** [comment]
To follow up on this, can you clarify how exactly you plan to use this?
We have an option `requires_grad_opt` in the tensor and model loaders that turn on gradient calculation, and `is_training_opt` for models.

See:
https://cambridge-iccs.github.io/FTorch/interface/torch_tensor_from_array.html
https://cambridge-iccs.github.io/FTorch/proc/torch_module_load.html

If either of those help.

**Comment by jatkinson1000** [comment]
Hi @nckxneoro52449 
Our latest update to FTorch involved an API change in an effort to prepare for further bringing the autograd feature into the Fortran side of codes.

Please see https://cambridge-iccs.github.io/FTorch/page/updates.html for a summary of this.

You can see the work that is going on to bring AutoGrad features at #139 and any input is welcome.

I will close this issue for now as it raises points that I have provided pointers for, but please do get in touch if there is anything specific you would like to see.
